src/audio/asrc/asrc_ipc3.c

```diff
@@ -978,20 +294,7 @@ static int asrc_reset(struct comp_dev *dev)
 {
 	struct comp_data *cd = comp_get_drvdata(dev);
 
-	comp_info(dev, "asrc_reset()");
-	comp_info(dev, "asrc_reset(), skew_min=%d, skew_max=%d", cd->skew_min,
-		  cd->skew_max);
-
-	/* If any resources feasible to stop */
-	if (cd->track_drift)
-		asrc_dai_stop_timestamp(cd);
-
-	/* Free the allocations those were done in prepare() */
-	asrc_release_buffers(cd->asrc_obj);
-	rfree(cd->asrc_obj);
-	rfree(cd->buf);
-	cd->asrc_obj = NULL;
-	cd->buf = NULL;
+	asrc_reset_general(cd, dev);
```

Comment:
so first a 1000+ line long file is created, then almost 700 lines get moved out...

Reply:
it is because I didn't have environment to test IPC3 version before, just want to keep the original code to reduce errors imported by me. Since Seppo help to enable the asrc test bench, I can convert IPC3 too. Thanks for reviewing fast, there will be a convert with both IPC3/IPC4 soon.